src/core/server/saved_objects/migrations/core/index_migrator.test.ts

```diff
@@ -454,23 +455,28 @@ function withIndex(
     opensearchClientMock.createSuccessTransportRequestPromise({
       task: 'zeid',
       _shards: { successful: 1, total: 1 },
-    })
+    } as opensearchtypes.ReindexResponse)
   );
   client.tasks.get.mockReturnValue(
-    opensearchClientMock.createSuccessTransportRequestPromise({ completed: true })
+    opensearchClientMock.createSuccessTransportRequestPromise({
+      completed: true,
+    } as opensearchtypes.TaskGetResponse)
   );
   client.search.mockReturnValue(
-    opensearchClientMock.createSuccessTransportRequestPromise(searchResult(0))
+    opensearchClientMock.createSuccessTransportRequestPromise(searchResult(0) as any)
```

Comment:
Do we have a type for this?

Reply:
Error:
```
Argument of type 'MockedTransportRequestPromise<ApiResponse<{ _scroll_id: number; _shards: { successful: number; total: number; }; hits: { hits: any; }; }, unknown>>' is not assignable to parameter of type 'TransportRequestPromise<ApiResponse<SearchResponse<unknown>, unknown>>'.
  The types returned by 'finally(...)' are incompatible between these types.
    Type 'Promise<ApiResponse<{ _scroll_id: number; _shards: { successful: number; total: number; }; hits: { hits: any; }; }, unknown>>' is not assignable to type 'Promise<ApiResponse<SearchResponse<unknown>, unknown>>'.
      Type 'ApiResponse<{ _scroll_id: number; _shards: { successful: number; total: number; }; hits: { hits: any; }; }, unknown>' is not assignable to type 'ApiResponse<SearchResponse<unknown>, unknown>'.
        Type '{ _scroll_id: number; _shards: { successful: number; total: number; }; hits: { hits: any; }; }' is missing the following properties from type 'SearchResponse<unknown>': took, timed_outts(2345)
```

Can't assign type to opensearchtypes.SearchResponse. This is because searchResult for the test is defined simply here:
```
const searchResult = (i: number) => ({
    _scroll_id: i,
    _shards: {
      successful: 1,
      total: 1,
    },
    hits: {
      hits: docs[i] || [],
    },
  });
```
While SearchResponse is defined as:
```
export interface SearchResponse<TDocument = unknown> {
  took: long
  timed_out: boolean
  _shards: ShardStatistics
  hits: SearchHitsMetadata<TDocument>
  aggregations?: Record<AggregateName, AggregationsAggregate>
  _clusters?: ClusterStatistics
  documents?: TDocument[]
  fields?: Record<string, any>
  max_score?: double
  num_reduce_phases?: long
  profile?: SearchProfile
  pit_id?: Id
  _scroll_id?: ScrollId
  suggest?: Record<SuggestionName, SearchSuggest<TDocument>[]>
  terminated_early?: boolean
}
```
SearchResponse is defined much more complicated and it is not needed for test purpose.